dev/service-registry/client-javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/registry/ServiceRegistryModule.scala

```diff
@@ -38,7 +40,8 @@ class ServiceRegistryModule(environment: Environment, configuration: Configurati
       Seq(
         bind[ServiceRegistryServiceLocator.ServiceLocatorConfig].toInstance(createDevServiceLocatorConfig),
         bind[ServiceRegistry].to(new ServiceRegistryClientProvider),
-        bind[ServiceLocator].to[ServiceRegistryServiceLocator]
+        bind[ServiceLocator].to[ServiceRegistryServiceLocator],
+        bind[DevModeSimpleServiceDiscovery].toProvider[DevModeSimpleServiceDiscoveryProvider].eagerly()
```

Comment:
TODO: also bind `SimpleServiceDiscovery` interface.

Reply:
Why do we need a `bind` to `DevModeSimpleServiceDiscovery`? Could we _just_ `bind[SimpleServiceDiscovery].toProvider[DevModeSimpleServiceDiscoveryProvider].eagerly()` and make `DevModeSimpleServiceDiscoveryProvider` a `Provider[SimpleServiceDiscovery]` ? 

Same question applies to the `instanceOf[DevModeSimpleServiceDiscovery]` below.

Reply:
I can't remember now why I did that binding in the first place, but I think it's obsolete.

Reply:
The cast is needed to call the `setServiceLocator` method (because injection is not available for Akka extensions).